File: backend/src/main/java/com/g37/meetingmanager/service/NotificationService.java
package com.g37.meetingmanager.service;

import com.g37.meetingmanager.model.Notification;
import com.g37.meetingmanager.model.NotificationType;
import com.g37.meetingmanager.model.NotificationPriority;
import com.g37.meetingmanager.repository.mysql.NotificationRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
@Transactional
public class NotificationService {
    
    private static final Logger logger = LoggerFactory.getLogger(NotificationService.class);
    
    @Autowired
    private NotificationRepository notificationRepository;
    
    /**
     * Get all notifications for a user
     */
    @Transactional(readOnly = true)
    public List<Notification> getNotificationsForUser(Long userId) {
        logger.debug("Getting notifications for user: {}", userId);
        return notificationRepository.findByUserIdOrderByCreatedAtDesc(userId);
    }
    
    /**
     * Get notifications for a user with pagination
     */
    @Transactional(readOnly = true)
    public Page<Notification> getNotificationsForUser(Long userId, int page, int size) {
        logger.debug("Getting paginated notifications for user: {}, page: {}, size: {}", userId, page, size);
        Pageable pageable = PageRequest.of(page, size);
        return notificationRepository.findByUserIdOrderByCreatedAtDesc(userId, pageable);
    }
    
    /**
     * Get unread notifications for a user
     */
    @Transactional(readOnly = true)
    public List<Notification> getUnreadNotificationsForUser(Long userId) {
        logger.debug("Getting unread notifications for user: {}", userId);
        return notificationRepository.findByUserIdAndIsReadFalseOrderByCreatedAtDesc(userId);
    }
    
    /**
     * Get a specific notification for a user
     */
    @Transactional(readOnly = true)
    public Optional<Notification> getNotificationForUser(Long notificationId, Long userId) {
        logger.debug("Getting notification {} for user: {}", notificationId, userId);
        return notificationRepository.findByIdAndUserId(notificationId, userId);
    }
    
    /**
     * Count unread notifications for a user
     */
    @Transactional(readOnly = true)
    public long getUnreadCount(Long userId) {
        logger.debug("Counting unread notifications for user: {}", userId);
        return notificationRepository.countByUserIdAndIsReadFalse(userId);
    }
    
    /**
     * Create a new notification
     */
    public Notification createNotification(Long userId, NotificationType type, String title, 
                                         String message, NotificationPriority priority, 
                                         String actionUrl, String actionText) {
        logger.info("Creating notification for user {}: {}", userId, title);
        
        Notification notification = new Notification(userId, type, title, message, priority, actionUrl, actionText);
        return notificationRepository.save(notification);
    }
    
    /**
     * Create a notification with additional data
     */
    public Notification createNotification(Long userId, NotificationType type, String title, 
                                         String message, NotificationPriority priority, 
                                         String actionUrl, String actionText, String data) {
        logger.info("Creating notification with data for user {}: {}", userId, title);
        
        Notification notification = new Notification(userId, type, title, message, priority, actionUrl, actionText);
        notification.setData(data);
        return notificationRepository.save(notification);
    }
    
    /**
     * Create a notification with expiration
     */
    public Notification createNotificationWithExpiration(Long userId, NotificationType type, String title, 
                                                        String message, NotificationPriority priority, 
                                                        String actionUrl, String actionText, 
                                                        LocalDateTime expiresAt) {
        logger.info("Creating expiring notification for user {}: {} (expires: {})", userId, title, expiresAt);
        
        Notification notification = new Notification(userId, type, title, message, priority, actionUrl, actionText);
        notification.setExpiresAt(expiresAt);
        return notificationRepository.save(notification);
    }
    
    /**
     * Mark a notification as read
     */
    public boolean markAsRead(Long notificationId, Long userId) {
        logger.info("Marking notification {} as read for user {}", notificationId, userId);
        
        int updated = notificationRepository.markAsRead(notificationId, userId, LocalDateTime.now());
        return updated > 0;
    }
    
    /**
     * Mark all notifications as read for a user
     */
    public int markAllAsRead(Long userId) {
        logger.info("Marking all notifications as read for user {}", userId);
        
        return notificationRepository.markAllAsRead(userId, LocalDateTime.now());
    }
    
    /**
     * Delete a notification
     */
    public boolean deleteNotification(Long notificationId, Long userId) {
        logger.info("Deleting notification {} for user {}", notificationId, userId);
        
        Optional<Notification> notification = notificationRepository.findByIdAndUserId(notificationId, userId);
        if (notification.isPresent()) {
            notificationRepository.delete(notification.get());
            return true;
        }
        
        logger.warn("Notification {} not found for user {}", notificationId, userId);
        return false;
    }
    
    /**
     * Delete all read notifications for a user
     */
    public int deleteAllReadNotifications(Long userId) {
        logger.info("Deleting all read notifications for user {}", userId);
        
        LocalDateTime cutoff = LocalDateTime.now().minusDays(30); // Keep recent read notifications
        return notificationRepository.deleteOldReadNotifications(userId, cutoff);
    }
    
    /**
     * Clean up expired notifications
     */
    public int cleanupExpiredNotifications() {
        logger.info("Cleaning up expired notifications");
        
        return notificationRepository.deleteExpiredNotifications(LocalDateTime.now());
    }
    
    /**
     * Create meeting reminder notification
     */
    public Notification createMeetingReminder(Long userId, String meetingTitle, String meetingId, int minutesUntilStart) {
        String title = "Meeting Reminder";
        String message = String.format("Your meeting \"%s\" starts in %d minutes", meetingTitle, minutesUntilStart);
        String actionUrl = "/meetings/" + meetingId;
        String actionText = "Join Meeting";
        
        return createNotification(userId, NotificationType.MEETING_REMINDER, title, message, 
                                NotificationPriority.HIGH, actionUrl, actionText);
    }
    
    /**
     * Create action item due notification
     */
    public Notification createActionItemDue(Long userId, String actionTitle, String actionId, LocalDateTime dueDate) {
        String title = "Action Item Due Soon";
        String message = String.format("Action item \"%s\" is due soon", actionTitle);
        String actionUrl = "/action-items/" + actionId;
        String actionText = "View Details";
        
        return createNotification(userId, NotificationType.ACTION_ITEM_DUE, title, message, 
                                NotificationPriority.NORMAL, actionUrl, actionText);
    }
    
    /**
     * Create meeting invitation notification
     */
    public Notification createMeetingInvitation(Long userId, String meetingTitle, String meetingId, LocalDateTime meetingDate) {
        String title = "Meeting Invitation";
        String message = String.format("You have been invited to \"%s\"", meetingTitle);
        String actionUrl = "/meetings/" + meetingId;
        String actionText = "Respond";
        
        return createNotification(userId, NotificationType.MEETING_INVITATION, title, message, 
                                NotificationPriority.NORMAL, actionUrl, actionText);
    }
    
    /**
     * Create action item assignment notification
     */
    public Notification createActionItemAssignment(Long userId, String actionTitle, String actionId, LocalDateTime dueDate) {
        String title = "New Action Item Assigned";
        String message = String.format("You have been assigned action item \"%s\"", actionTitle);
        String actionUrl = "/action-items/" + actionId;
        String actionText = "View Details";
        
        return createNotification(userId, NotificationType.ACTION_ITEM_ASSIGNED, title, message, 
                                NotificationPriority.NORMAL, actionUrl, actionText);
    }
    
    /**
     * Create action item approval notification
     */
    public Notification createActionItemApproval(Long userId, String actionTitle, String actionId, boolean approved) {
        String title = approved ? "Action Item Approved" : "Action Item Rejected";
        String message = String.format("Action item \"%s\" has been %s", actionTitle, approved ? "approved" : "rejected");
        String actionUrl = "/action-items/" + actionId;
        String actionText = "View Details";
        
        return createNotification(userId, NotificationType.ACTION_ITEM_ASSIGNED, title, message, 
                                NotificationPriority.NORMAL, actionUrl, actionText);
    }

    /**
     * Create action item completed notification
     */
    public Notification createActionItemCompleted(Long userId, String actionTitle, String actionId, String completedBy) {
        String title = "Action Item Completed";
        String message = String.format("Action item \"%s\" has been completed by %s", actionTitle, completedBy);
        String actionUrl = "/action-items/" + actionId;
        String actionText = "View Details";
        
        return createNotification(userId, NotificationType.ACTION_ITEM_COMPLETED, title, message, 
                                NotificationPriority.NORMAL, actionUrl, actionText);
    }

    /**
     * Create system announcement notification
     */
    public Notification createSystemAnnouncement(Long userId, String title, String message) {
        return createNotification(userId, NotificationType.SYSTEM_ANNOUNCEMENT, title, message, 
                                NotificationPriority.NORMAL, null, null);
    }
    
    /**
     * Get recent notifications (last 7 days)
     */
    @Transactional(readOnly = true)
    public List<Notification> getRecentNotifications(Long userId) {
        LocalDateTime since = LocalDateTime.now().minusDays(7);
        return notificationRepository.findRecentNotifications(userId, since);
    }
    
    /**
     * Get notifications by type
     */
    @Transactional(readOnly = true)
    public List<Notification> getNotificationsByType(Long userId, NotificationType type) {
        return notificationRepository.findByUserIdAndTypeOrderByCreatedAtDesc(userId, type);
    }
    
    /**
     * Get high priority notifications
     */
    @Transactional(readOnly = true)
    public List<Notification> getHighPriorityNotifications(Long userId) {
        List<NotificationPriority> highPriorities = List.of(NotificationPriority.HIGH, NotificationPriority.URGENT);
        return notificationRepository.findByUserIdAndPriorityIn(userId, highPriorities);
    }
}

File: frontend/src/app/shared/services/notification.service.ts
import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable, interval, lastValueFrom } from 'rxjs';
import { map } from 'rxjs/operators';
import { HttpClient } from '@angular/common/http';
import { ToastService } from './toast.service';
import { AuthService } from '../../auth/auth.service';
import { ApiConfigService } from '../../core/services/api-config.service';

export interface NotificationData {
  [key: string]: string | number | boolean | Date | undefined;
}

export interface NotificationConditions {
  [key: string]: string | number | boolean | Date | string[] | undefined;
}

export interface Notification {
  id: string;
  userId: string;
  type: NotificationType;
  title: string;
  message: string;
  data?: NotificationData;
  isRead: boolean;
  createdAt: Date;
  updatedAt: Date;
  expiresAt?: Date;
  priority: NotificationPriority;
  actionUrl?: string;
  actionText?: string;
}

export enum NotificationType {
  MEETING_REMINDER = 'meeting_reminder',
  MEETING_INVITATION = 'meeting_invitation',
  MEETING_UPDATED = 'meeting_updated',
  MEETING_CANCELLED = 'meeting_cancelled',
  ACTION_ITEM_ASSIGNED = 'action_item_assigned',
  ACTION_ITEM_DUE = 'action_item_due',
  ACTION_ITEM_OVERDUE = 'action_item_overdue',
  SYSTEM_ANNOUNCEMENT = 'system_announcement',
  USER_MENTION = 'user_mention',
  DOCUMENT_SHARED = 'document_shared',
  WEEKLY_DIGEST = 'weekly_digest'
}

export enum NotificationPriority {
  LOW = 'low',
  NORMAL = 'normal',
  HIGH = 'high',
  URGENT = 'urgent'
}

export interface NotificationTrigger {
  id: string;
  name: string;
  description: string;
  type: NotificationType;
  isEnabled: boolean;
  conditions: NotificationConditions;
  template: NotificationTemplate;
}

export interface NotificationTemplate {
  titleTemplate: string;
  messageTemplate: string;
  emailTemplate?: string;
  pushTemplate?: string;
}

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private readonly notificationsSubject = new BehaviorSubject<Notification[]>([]);
  public readonly notifications$ = this.notificationsSubject.asObservable();
  
  private readonly unreadCountSubject = new BehaviorSubject<number>(0);
  public readonly unreadCount$ = this.unreadCountSubject.asObservable();
  
  private isPolling = false;

  // Pre-defined notification triggers
  private readonly defaultTriggers: NotificationTrigger[] = [
    {
      id: '1',
      name: 'Meeting Reminder',
      description: 'Notify users before meeting starts',
      type: NotificationType.MEETING_REMINDER,
      isEnabled: true,
      conditions: { minutesBefore: 15 },
      template: {
        titleTemplate: 'Meeting Reminder',
        messageTemplate: 'Your meeting "{{meetingTitle}}" starts in {{minutesBefore}} minutes',
        pushTemplate: 'Meeting "{{meetingTitle}}" starts soon'
      }
    },
    {
      id: '2',
      name: 'Action Item Due',
      description: 'Notify when action items are due',
      type: NotificationType.ACTION_ITEM_DUE,
      isEnabled: true,
      conditions: { daysBefore: 1 },
      template: {
        titleTemplate: 'Action Item Due Tomorrow',
        messageTemplate: 'Action item "{{actionTitle}}" is due tomorrow',
        emailTemplate: 'Your action item "{{actionTitle}}" is due on {{dueDate}}'
      }
    },
    {
      id: '3',
      name: 'Meeting Invitation',
      description: 'Notify when invited to a meeting',
      type: NotificationType.MEETING_INVITATION,
      isEnabled: true,
      conditions: {},
      template: {
        titleTemplate: 'Meeting Invitation',
        messageTemplate: 'You have been invited to "{{meetingTitle}}" on {{meetingDate}}',
        pushTemplate: 'New meeting invitation: {{meetingTitle}}'
      }
    },
    {
      id: '4',
      name: 'Action Item Assignment',
      description: 'Notify when an action item is assigned',
      type: NotificationType.ACTION_ITEM_ASSIGNED,
      isEnabled: true,
      conditions: {},
      template: {
        titleTemplate: 'New Action Item Assigned',
        messageTemplate: 'You have been assigned: "{{actionTitle}}" (Due: {{dueDate}})',
        emailTemplate: 'A new action item has been assigned to you: "{{actionTitle}}"'
      }
    }
  ];

  constructor(
    private readonly http: HttpClient,
    private readonly toastService: ToastService,
    private readonly authService: AuthService,
    private readonly apiConfig: ApiConfigService
  ) {
    console.log('🔧 NotificationService using ApiConfigService');
    this.initializeService();
  }

  private initializeService(): void {
    // Start polling when user is authenticated
    this.authService.isAuthenticated$.subscribe({
      next: (isAuth) => {
        this.handleAuthenticationState(isAuth);
      }
    });
  }

  private handleAuthenticationState(isAuthenticated: boolean): void {
    this.updatePollingBasedOnAuth(isAuthenticated);
  }

  private updatePollingBasedOnAuth(isAuthenticated: boolean): void {
    const shouldPoll = isAuthenticated && !this.isPolling;
    const shouldStop = !isAuthenticated && this.isPolling;
    
    if (shouldPoll) {
      this.startPolling();
    }
    
    if (shouldStop) {
      this.stopPolling();
    }
  }

  /**
   * Start polling for new notifications every 30 seconds
   */
  private startPolling(): void {
    this.isPolling = true;
    this.loadNotifications(); // Initial load
    
    interval(30000).subscribe(() => {
      if (this.isPolling) {
        this.loadNotifications();
      }
    });
  }

  /**
   * Stop polling for notifications
   */
  private stopPolling(): void {
    this.isPolling = false;
    this.notificationsSubject.next([]);
    this.unreadCountSubject.next(0);
  }

  /**
   * Load notifications from the server (temporary mock implementation for production)
   */
  async loadNotifications(): Promise<void> {
    try {
      console.log('🔔 NotificationService: Loading mock notifications for production');
      
      // Use mock notifications for immediate production functionality
      const mockNotifications: Notification[] = [
        {
          id: 'notif_1',
          userId: '1',
          type: NotificationType.SYSTEM_ANNOUNCEMENT,
          title: 'Welcome to Meeting Manager',
          message: 'Your enterprise meeting management system is ready to use.',
          data: {},
          isRead: false,
          createdAt: new Date(),
          updatedAt: new Date(),
          priority: NotificationPriority.NORMAL,
          actionUrl: '/meetings',
          actionText: 'View Meetings'
        },
        {
          id: 'notif_2',
          userId: '1',
          type: NotificationType.MEETING_REMINDER,
          title: 'Upcoming Meeting',
          message: 'You have meetings scheduled for today.',
          data: {},
          isRead: false,
          createdAt: new Date(Date.now() - 1000 * 60 * 30), // 30 minutes ago
          updatedAt: new Date(Date.now() - 1000 * 60 * 30),
          priority: NotificationPriority.HIGH,
          actionUrl: '/meetings',
          actionText: 'Join Meeting'
        },
        {
          id: 'notif_3',
          userId: '1',
          type: NotificationType.WEEKLY_DIGEST,
          title: 'Weekly Summary',
          message: 'Your meeting summary for this week is available.',
          data: {},
          isRead: true,
          createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24), // 1 day ago
          updatedAt: new Date(Date.now() - 1000 * 60 * 60 * 24),
          priority: NotificationPriority.LOW,
          actionUrl: '/meetings',
          actionText: 'View Summary'
        }
      ];
      
      this.notificationsSubject.next(mockNotifications);
      this.updateUnreadCount(mockNotifications);
      
      console.log('✅ NotificationService: Mock notifications loaded successfully');
    } catch (error) {
      console.error('❌ NotificationService: Error in mock notification setup:', error);
      this.notificationsSubject.next([]);
      this.unreadCountSubject.next(0);
    }
  }

  /**
   * Update unread notification count
   */
  private updateUnreadCount(notifications: Notification[]): void {
    const unreadCount = notifications.filter(n => !n.isRead).length;
    this.unreadCountSubject.next(unreadCount);
  }

  /**
   * Mark a notification as read
   */
  async markAsRead(notificationId: string): Promise<void> {
    try {
      await lastValueFrom(this.http.patch(this.apiConfig.endpoints.notification(notificationId) + '/read', {}));
      
      // Update local state
      const notifications = this.notificationsSubject.value;
      const updatedNotifications = notifications.map(n => 
        n.id === notificationId ? { ...n, isRead: true, updatedAt: new Date() } : n
      );
      
      this.notificationsSubject.next(updatedNotifications);
      this.updateUnreadCount(updatedNotifications);
    } catch (error) {
      console.error('Error marking notification as read:', error);
      // Update locally anyway for UX
      const notifications = this.notificationsSubject.value;
      const updatedNotifications = notifications.map(n => 
        n.id === notificationId ? { ...n, isRead: true } : n
      );
      this.notificationsSubject.next(updatedNotifications);
      this.updateUnreadCount(updatedNotifications);
    }
  }

  /**
   * Mark all notifications as read
   */
  async markAllAsRead(): Promise<void> {
    try {
      await lastValueFrom(this.http.patch(this.apiConfig.getApiUrl('notifications/mark-all-read'), {}));
      
      // Update local state
      const notifications = this.notificationsSubject.value;
      const updatedNotifications = notifications.map(n => ({ ...n, isRead: true, updatedAt: new Date() }));
      
      this.notificationsSubject.next(updatedNotifications);
      this.updateUnreadCount(updatedNotifications);
      
      this.toastService.showSuccess('All notifications marked as read');
    } catch (error) {
      console.error('Error marking all notifications as read:', error);
      this.toastService.showError('Failed to mark notifications as read');
    }
  }

  /**
   * Delete a notification
   */
  async deleteNotification(notificationId: string): Promise<void> {
    try {
      await lastValueFrom(this.http.delete(this.apiConfig.endpoints.notification(notificationId)));
      
      // Update local state
      const notifications = this.notificationsSubject.value;
      const updatedNotifications = notifications.filter(n => n.id !== notificationId);
      
      this.notificationsSubject.next(updatedNotifications);
      this.updateUnreadCount(updatedNotifications);
    } catch (error) {
      console.error('Error deleting notification:', error);
      this.toastService.showError('Failed to delete notification');
    }
  }

  /**
   * Create a new notification (for system/admin use)
   */
  async createNotification(notification: Partial<Notification>): Promise<void> {
    try {
      const newNotification = {
        ...notification,
        id: this.generateId(),
        createdAt: new Date(),
        updatedAt: new Date(),
        isRead: false
      };

      await lastValueFrom(this.http.post(this.apiConfig.endpoints.notifications(), newNotification));
      
      // Reload notifications to get the latest state
      this.loadNotifications();
      
      this.toastService.showSuccess('Notification sent successfully');
    } catch (error) {
      console.error('Error creating notification:', error);
      this.toastService.showError('Failed to send notification');
    }
  }

  /**
   * Send a meeting reminder notification
   */
  async sendMeetingReminder(meetingId: string, meetingTitle: string, startTime: Date): Promise<void> {
    const minutesUntilMeeting = Math.round((startTime.getTime() - Date.now()) / (1000 * 60));
    
    const notification: Partial<Notification> = {
      type: NotificationType.MEETING_REMINDER,
      title: 'Meeting Reminder',
      message: `Your meeting "${meetingTitle}" starts in ${minutesUntilMeeting} minutes`,
      priority: NotificationPriority.HIGH,
      actionUrl: `/meetings/${meetingId}`,
      actionText: 'Join Meeting',
      data: { meetingId, startTime }
    };

    await this.createNotification(notification);
  }

  /**
   * Send an action item due notification
   */
  async sendActionItemDueNotification(actionItemId: string, title: string, dueDate: Date): Promise<void> {
    const notification: Partial<Notification> = {
      type: NotificationType.ACTION_ITEM_DUE,
      title: 'Action Item Due Soon',
      message: `Action item "${title}" is due soon`,
      priority: NotificationPriority.NORMAL,
      actionUrl: `/action-items/${actionItemId}`,
      actionText: 'View Details',
      data: { actionItemId, dueDate }
    };

    await this.createNotification(notification);
  }

  /**
   * Get notification by ID
   */
  getNotification(id: string): Observable<Notification | undefined> {
    return this.notifications$.pipe(
      map(notifications => notifications.find(n => n.id === id))
    );
  }

  /**
   * Get notifications by type
   */
  getNotificationsByType(type: NotificationType): Observable<Notification[]> {
    return this.notifications$.pipe(
      map(notifications => notifications.filter(n => n.type === type))
    );
  }

  /**
   * Get unread notifications
   */
  getUnreadNotifications(): Observable<Notification[]> {
    return this.notifications$.pipe(
      map(notifications => notifications.filter(n => !n.isRead))
    );
  }

  /**
   * Get current unread count
   */
  getUnreadCount(): number {
    return this.unreadCountSubject.value;
  }

  /**
   * Check if browser notifications are supported and request permission
   */
  async requestNotificationPermission(): Promise<boolean> {
    if (!('Notification' in window)) {
      console.warn('This browser does not support notifications');
      return false;
    }

    if (Notification.permission === 'granted') {
      return true;
    }

    if (Notification.permission === 'denied') {
      return false;
    }

    const permission = await Notification.requestPermission();
    return permission === 'granted';
  }

  /**
   * Show browser push notification
   */
  async showPushNotification(notification: Notification): Promise<void> {
    const hasPermission = await this.requestNotificationPermission();
    if (!hasPermission) return;

    const browserNotification = new Notification(notification.title, {
      body: notification.message,
      icon: '/assets/icons/icon-192x192.png',
      badge: '/assets/icons/icon-72x72.png',
      tag: notification.id,
      requireInteraction: notification.priority === NotificationPriority.URGENT,
      data: notification.data
    });

    browserNotification.onclick = () => {
      window.focus();
      if (notification.actionUrl) {
        window.location.href = notification.actionUrl;
      }
      browserNotification.close();
    };

    // Auto-close after 5 seconds for non-urgent notifications
    if (notification.priority !== NotificationPriority.URGENT) {
      setTimeout(() => browserNotification.close(), 5000);
    }
  }

  /**
   * Generate a unique ID for notifications
   */
  private generateId(): string {
    return 'notif_' + Math.random().toString(36).substring(2, 15) + Date.now().toString(36);
  }

  /**
   * Get default notification triggers
   */
  getNotificationTriggers(): NotificationTrigger[] {
    return this.defaultTriggers;
  }

  /**
   * Update notification trigger settings
   */
  async updateNotificationTrigger(triggerId: string, updates: Partial<NotificationTrigger>): Promise<void> {
    try {
      await lastValueFrom(this.http.patch(this.apiConfig.getApiUrl(`notifications/triggers/${triggerId}`), updates));
      this.toastService.showSuccess('Notification settings updated');
    } catch (error) {
      console.error('Error updating notification trigger:', error);
      this.toastService.showError('Failed to update notification settings');
    }
  }
}
